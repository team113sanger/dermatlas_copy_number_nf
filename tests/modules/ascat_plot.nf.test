
nextflow_process {

    name "Test Process CREATE_FREQUENCY_PLOTS"
    script "modules/ascat.nf"
    process "CREATE_FREQUENCY_PLOTS"

    test("Should run without failures") {

        when {
        params {
            load("$baseDir/tests/testdata/test_params.json")
        }
            process {
                """
                input[0] = file('$baseDir/tests/testdata/one_tumor_per_patient.tsv')
                input[1] = file('$baseDir/tests/testdata/sample_purity_ploidy.tsv')
                input[2] = file('$baseDir/tests/testdata/sex2chr.txt')
                input[3] = "test"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
