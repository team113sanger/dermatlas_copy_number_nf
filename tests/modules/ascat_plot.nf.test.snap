{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "analysis_type": "one_tumor_per_patient",
                        "plot_dir": "one_per_patient"
                    },
                    "test_cn-loh.tsv:md5,340475db7aea3af3d5173d5e9390b904"
                ]
            ],
            [
                [
                    {
                        "analysis_type": "one_tumor_per_patient",
                        "plot_dir": "one_per_patient"
                    },
                    "test_CNfreq.tsv:md5,89fdaa4849518f14f10fb30b7ead7d9b"
                ]
            ],
            [
                [
                    {
                        "analysis_type": "one_tumor_per_patient",
                        "plot_dir": "one_per_patient"
                    },
                    "test_cn-loh_segments.tsv:md5,218645f4ab0686d694bb21261073345b"
                ]
            ],
            [
                [
                    {
                        "analysis_type": "one_tumor_per_patient",
                        "plot_dir": "one_per_patient"
                    },
                    "test_segments.tsv:md5,f4b548e21ed6a3a1e50b1cd33ff8bfa6"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-12-13T17:50:48.932467"
    }
}